src/code/TextTableParser.cs

```diff
@@ -0,0 +1,505 @@
+            }
+            else
+            {
+                ColumnInfoList = GetColumnList(Lines.Count, SpaceArray);
+            }
+
+            string[]columnHeaders;
+            if (NoHeader)
+            {
+                columnHeaders = new string[ColumnInfoList.Count];
+                for(int i = 0; i < ColumnInfoList.Count; i++)
+                {
+                    columnHeaders[i] = string.Format("Column_{0:00}", i+1);
+                }
+            }
+            else
+            {
+                columnHeaders = GetHeaderColumns(ColumnInfoList, Lines[HeaderLine]);
+            }
+
+            // WriteObject(ColumnInfoList);
+            // WriteObject(columnHeaders);
+
+            WriteDebug(string.Format("SpaceArrayLength: {0}", SpaceArray.Length));
+            WriteDebug(spaceRepresentation.TrimEnd());
+            
+            for(int i = HeaderLine+1; i < Lines.Count; i++)
+            {
+                if (AsPsObject)
+                {
+                    WriteObject(GetPsObject(ColumnInfoList, Lines[i], columnHeaders));
+                }
+                else
+                {
+                    WriteObject(GetRow(ColumnInfoList, Lines[i], columnHeaders));
+                }
+
+            }
+
+        }
+        */
+
+        private string GetSpaceRepresentation(int count, int[] spaceArray)
+        {
+            char[] spaceChars = new char[spaceArray.Length];
+            for(int i = 0; i < spaceArray.Length; i++)
+            {
+                if (spaceArray[i] == count)
+                {
+                    spaceChars[i] = 'S';
+                }
+                else
+                {
+                    spaceChars[i] = ' ';
+                }
+            }
+            return new string(spaceChars);
+        }
+
+        public PSObject GetPsObject(List<ColumnInfo> cInfos, string line, string[] columnHeaders)
+        {
+            PSObject o = new PSObject();
+            if (Typename != null)
+            {
+                foreach (string t in Typename)
+                {
+                    o.TypeNames.Insert(0, t);
+                }
+            }
+            object[]data = GetObjectColumnData(cInfos, line);
+            Debug.Assert(data.Length == columnHeaders.Length);
+            for(int i = 0; i < cInfos.Count; i++)
+            {
+                o.Properties.Add(new PSNoteProperty(columnHeaders[i], data[i]));
+            }
+            return o;
+        }
+
+        public string GetJson(List<ColumnInfo> cInfos, string line, string[] columnHeaders)
+        {
+            string[]data = GetStringColumnData(cInfos, line);
+            Debug.Assert(data.Length == columnHeaders.Length);
+            string[]dataWithHeader = new string[data.Length];
+            for(int j = 0; j < data.Length; j++)
+            {
+                if (data[j] is string)
+                {
+                    // TODO: json encode the string

```

Comment:
I don't understand this TODO, isn't line 315 encoding as json?

Reply:
yah - I missed cleaning this up